src/yunohost/tools.py

```diff
@@ -162,6 +162,28 @@ def tools_maindomain(auth, new_domain=None):
         logger.warning("%s" % e, exc_info=1)
         raise MoulinetteError(errno.EPERM, m18n.n('maindomain_change_failed'))
 
+    # Set hostname
+    pretty_hostname = "(YunoHost/%s)" % new_domain
+    commands = [
+        "sudo hostnamectl --static    set-hostname".split() + [new_domain],
+        "sudo hostnamectl --transient set-hostname".split() + [new_domain],
+        "sudo hostnamectl --pretty    set-hostname".split() + [pretty_hostname]
```

Comment:
I think hostnamectl is precisely designed to make change persitent (unlike hostname and other ways). Also this is already what Doctor Cube does : https://github.com/labriqueinternet/doctorcube_ynh/blob/6d8580c0216c925cdd2ea435a23a9f0408f6a471/scripts/install#L18-L20